Added Login and Register Page

KMSZamil opened the pull request:
I have updated two pages "Login" and "Register". Accessible from /login and /register.

![image](https://github.com/user-attachments/assets/ca64e7ae-4a63-4d83-a622-59ad54d8b22a)

![image](https://github.com/user-attachments/assets/a0cc57b3-8f12-4799-8f0e-ebeef24f4901)

skfarhad reviewed (changes requested):
Could you change the color scheme so that it's consistent with this design?
![Uploading teamwork_dashboard.png…]()

KMSZamil commented:
The color is turned to white.

![image](https://github.com/user-attachments/assets/d296194e-c66d-4da3-94bc-0818ffee8c70)

Hopefully change the design as designed

KMSZamil commented:
Added new login screen

![image](https://github.com/user-attachments/assets/26302ff3-5ca1-4566-815a-63300ed1788b)

Added new register screen

![image](https://github.com/user-attachments/assets/2039c582-9365-4334-9b74-698f1edb25fd)

Muttakee31 reviewed (approved):
LGTM. The merge conflict seems to be a simple one.
Just two suggestions.

1. We can remove the Facebook logo if it is not being used for the moment.
2. We may integrate an icon library like font awesome instead of using SVG images. It depends on the Figma design. if there are a significant amount of icons, we can adopt that.

Shabab007 commented on web_app/hazard-reporting-map/components/LoginForm.jsx:
```diff
@@ -0,0 +1,102 @@
+            if (!!response.error) {
+                setError(response.error.message);
+            } else {
+                //router.push("/");
```
remove comments

Shabab007 commented on web_app/hazard-reporting-map/components/LoginForm.jsx:
```diff
@@ -0,0 +1,102 @@
+                //router.push("/");
+            }
+        } catch (error) {
+            //setError(error.message);
```
remove comment

Shabab007 commented on web_app/hazard-reporting-map/components/RegistrationForm.jsx:
```diff
@@ -0,0 +1,105 @@
+        <>
+            {error && <div className="text-2xl text-center text-red-500">{error}</div>}
+            <form className="login-form" onSubmit={onSubmit}>
+                <div>
```
use mapping array for repeated elements

Shabab007 commented on web_app/hazard-reporting-map/package.json:
```diff
@@ -9,10 +9,12 @@
     "lint": "next lint"
   },
   "dependencies": {
```
you havent use these two packages . please remove it if its not necessary

KMSZamil replied:
I have removed unused packages

Shabab007 commented:
Design feedbacks according to figma:
1. header title "System" should be in next line and in center
2. place holders are not given
3. Fonts colors are not matching 
4. no picture in right side
5. gap between the login and google sign in button not correct 
6. i agree not aligned . highlighted texts not underline
7. on register page thana, org etc input dropdowns and placeholders not added

KMSZamil commented:
Resolved as said

Shabab007 commented on web_app/hazard-reporting-map/components/RegistrationForm.jsx:
```diff
@@ -41,33 +41,49 @@ const RegistrationForm = () => {
                 <div>
-                    <label htmlFor="fname">Full Name</label>
-                    <input type="text" name="fname" id="fname" />
+                    <label htmlFor="fname" className="text-gray-600">
```
only Fullname , mobile and email field are in the register form now according to update

Shabab007 commented on web_app/hazard-reporting-map/components/RegistrationForm.jsx:
```diff
@@ -41,33 +41,49 @@ const RegistrationForm = () => {
-                    <input type="text" name="mobile" id="mobile" />
+                    <label htmlFor="mobile" className="text-gray-600">
+                        Mobile No.
+                    </label>
```
add email field and validation also

Shabab007 commented on web_app/hazard-reporting-map/components/SocialLogin.jsx:
```diff
@@ -0,0 +1,36 @@
+const SocialLogins = ({ mode }) => {
+    const handleAuth = (e) => {
+        if (e === "google") {
+            signIn("google", { callbackUrl: "http://localhost:3000/" });
```
try not to use hard coded values use env for callbackUrl or use variable instead

Shabab007 commented on web_app/hazard-reporting-map/styles/globals.css:
```diff
@@ -2,36 +2,96 @@
 @tailwind utilities;
 
-:root {
+/* :root {
```
remove comments

Shabab007 commented:
Lastly , Resolve conflicts

Shabab007 commented on web_app/hazard-reporting-map/components/RegistrationForm.jsx:
```diff
@@ -0,0 +1,111 @@
+                        name="mobile"
+                        id="mobile"
+                        placeholder="Enter your mobile number"
+                        pattern="\d{11}"
```
need to add mobile number validation

Shabab007 commented on web_app/hazard-reporting-map/components/RegistrationForm.jsx:
```diff
@@ -0,0 +1,111 @@
+                <div>
+                    <label htmlFor="email" className="text-gray-600">
+                        Email
+                    </label>
```
email validation is missing

Shabab007 commented:
Please remove the old pics and upload updated pics or video so that we can look at the updated design

KMSZamil commented:
Updated in the first comment @Shabab007 bhai

Shabab007 commented:
@KMSZamil  design looks good . 
1. only top margin or space of the title not matching . in figma there is much space on the top compared to your pic 
2. Please add email and mobile number validation for register input. if you need help let me know

KMSZamil commented:
@Shabab007 bhai, Updated as requested

![image](https://github.com/user-attachments/assets/5e222086-c60a-439d-9eff-caad8cecf417)

![image](https://github.com/user-attachments/assets/59e708f1-830b-479f-924f-85ecefe3cce6)

Shabab007 commented on web_app/hazard-reporting-map/styles/globals.css:
```diff
@@ -4,6 +4,7 @@
 
 
 body {
+  font-family: 'Manrope', sans-serif;
```
there is a way to incorporate font in nextjs and tailwind. please correct that

skfarhad reviewed (approved):
Looks good!
